compare/json.go

```diff
@@ -0,0 +1,83 @@
+package compare
+
+import (
+	"encoding/json"
+
+	"github.com/yudai/gojsondiff"
+	"github.com/yudai/gojsondiff/formatter"
+)
+
+// JSON compares two json strings, processes them to handle wild cards,
+func JSON(a []byte, b []byte) (string, error) {
+	differ := gojsondiff.New()
+	d, err := differ.Compare(a, b)
+	if err != nil {
+		return "", err
+	}
+
+	// filter the fields we do not want
+	filteredDiffer := jsondiff{
+		deltas: filter(d.Deltas(), filterWildcards),
+	}
+
+	if filteredDiffer.Modified() {
+		var aJSON map[string]interface{}
+		json.Unmarshal(a, &aJSON)
```

Comment:
should we handle this err?

Reply:
also i think `aJSON` will be `nil`, Unmarshalling `a` into `nil` would cause panic

Reply:
this can be safely skipped I think, the differ unmarshals the JSON in differ.Compare(), and if that failed the error would have returned